blog/2020-02-16-debezium-on-rds.adoc

```diff
@@ -0,0 +1,226 @@
+
+== Debezium at Delhivery
+
+We work in the logistics landscape and hence most of the software we write is focused around state - status changes of a shipment, tracking location updates, collecting real time data and reacting to it. The most common place where you might find "state" in any software architecture is the database. We maintain all of our transactional data primarily in relational database systems (specifically PostgreSQL) as different databases for different services within the organisation. There is a need to allow efficient and near-real time analysis of the transactional data across all the different data sources to allow surfacing insights and looking at the big picture of how the organisation is doing and to make data driven decisions.
+
+To solve the above goal, we are using Debezium to perform Change Data Capture on our transactional data to make it available in Kafka, our choice of message broker. Once that data is available in Kafka we can do one or all of the following:
+
+* Perform streaming joins or data enrichment across change streams of different relational tables (maybe even from different databases or services altogether. eg. Enriching shipments with trip and vehicle data)
+* Creating domain events from change streams for consumption by downstream services (eg. Creating an aggregate message with order, shipment and product information from three different change streams)
+* Moving the change stream data into a data lake to allow for disaster recovery or replaying part of the data
+* Complex event processing to generate real time metrics and power dashboards (eg. Live count of items in transit, average trip time within each region etc.)
+
+Debezium makes all of those above use-cases possible and very easy to build by providing a common platform and framework to connect our existing data sources like MongoDB, PostgreSQL or MySQL.
+
+This article is going to share how to skip unparseable records from PostgreSQL's WAL when using Debezium until https://issues.redhat.com/browse/DBZ-1760[DBZ-1760] is solved. Additionally, it is going to share our learnings with using Debezium on AWS RDS (AWS's managed database service) and hopefully help transfer some knowledge we've gained in that process.
```

Comment:
These two sentences should be swapped, as per the re-ordering of the following sections.

Reply:
Nice catch.